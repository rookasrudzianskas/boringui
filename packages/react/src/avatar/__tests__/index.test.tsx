import React from "react";
import {mount, render, shallow} from "enzyme";

import Avatar from "../index";

interface IconProps {
  className?: string;
}

const Icon: React.FC<IconProps> = ({className = ""}) => <svg className={className} />;

describe("Avatar", () => {
  it("should support square and circle", () => {
    const circle = shallow(<Avatar />);

    expect(() => circle.unmount()).not.toThrow();
    const square = shallow(<Avatar squared />);

    expect(() => square.unmount()).not.toThrow();
  });

  it("should render text element", () => {
    const imageAvatar = render(<Avatar />);

    expect(imageAvatar).toMatchSnapshot();
    const textAvatar = render(<Avatar text="text" />);

    expect(textAvatar).toMatchSnapshot();
  });

  it("should work with different colors", () => {
    const wrapper = mount(
      <div>
        <Avatar color="primary" />
        <Avatar color="secondary" />
        <Avatar color="success" />
        <Avatar color="warning" />
        <Avatar color="error" />
      </div>,
    );

    expect(wrapper.html()).toMatchSnapshot();
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it("should work with a gradient color", () => {
    const wrapper = mount(<Avatar color="gradient" />);

    expect(wrapper.html()).toMatchSnapshot();
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it("should work with pointer", () => {
    const wrapper = mount(<Avatar pointer />);

    expect(wrapper.html()).toMatchSnapshot();
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it("should work with zoomed", () => {
    const wrapper = mount(<Avatar zoomed />);

    expect(wrapper.html()).toMatchSnapshot();
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it("should render correctly with an icon", () => {
    const wrapper = mount(<Avatar icon={<Icon />} />);

    expect(wrapper.html()).toMatchSnapshot();
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it("should omit long chars automatically", () => {
    const avatar = mount(<Avatar text="loremipsumtextlarge" />);
    const text = avatar.find(".nextui-avatar-text").text();

    expect(text.length).toBeLessThan(4);
  });

  it("stacked should be work", () => {
    const avatar = shallow(
      <Avatar stacked src="https://i.pravatar.cc/300?u=a042581f4e29026705d" />,
    );

    expect(() => avatar.unmount()).not.toThrow();
  });

  it("should work correctly with different sizes", () => {
    const wrapper = mount(
      <div>
        <Avatar size="xs" />
        <Avatar size="sm" />
        <Avatar size="md" />
        <Avatar size="lg" />
        <Avatar size="xl" />
      </div>,
    );

    expect(wrapper.html()).toMatchSnapshot();
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it("should render component of a specified size", () => {
    const avatar = render(<Avatar css={{size: 20}} />);

    expect(avatar).toMatchSnapshot();
  });
});
