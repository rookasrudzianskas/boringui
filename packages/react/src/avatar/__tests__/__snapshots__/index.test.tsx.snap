// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should render component of a specified size 1`] = `
LoadedCheerio {
  "0": <span
    class="nextui-c-gEEnVX nextui-c-PJLV nextui-c-gEEnVX-qXFXw-color-default nextui-c-gEEnVX-duCSmO-textColor-default nextui-c-gEEnVX-iHugVF-size-md nextui-c-gEEnVX-krOtXB-borderWeight-normal nextui-c-gEEnVX-bAEoQm-rounded-true nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-gEEnVX-ihFPSGE-css only-text-avatar"
    data-state="ready"
  >
    <span
      class="nextui-avatar-bg"
    />
  </span>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": Array [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Avatar should render correctly with an icon 1`] = `"<span class=\\"nextui-c-gEEnVX nextui-c-PJLV nextui-c-gEEnVX-qXFXw-color-default nextui-c-gEEnVX-duCSmO-textColor-default nextui-c-gEEnVX-iHugVF-size-md nextui-c-gEEnVX-krOtXB-borderWeight-normal nextui-c-gEEnVX-bAEoQm-rounded-true nextui-c-PJLV-gulvcB-isFocusVisible-false only-text-avatar\\" data-state=\\"ready\\"><span class=\\"nextui-avatar-bg\\"></span><span class=\\"nextui-avatar-icon\\"><svg class=\\"\\"></svg></span></span>"`;

exports[`Avatar should render text element 1`] = `
LoadedCheerio {
  "0": <span
    class="nextui-c-gEEnVX nextui-c-PJLV nextui-c-gEEnVX-qXFXw-color-default nextui-c-gEEnVX-duCSmO-textColor-default nextui-c-gEEnVX-iHugVF-size-md nextui-c-gEEnVX-krOtXB-borderWeight-normal nextui-c-gEEnVX-bAEoQm-rounded-true nextui-c-PJLV-gulvcB-isFocusVisible-false only-text-avatar"
    data-state="ready"
  >
    <span
      class="nextui-avatar-bg"
    />
  </span>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": Array [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Avatar should render text element 2`] = `
LoadedCheerio {
  "0": <span
    class="nextui-c-gEEnVX nextui-c-PJLV nextui-c-gEEnVX-qXFXw-color-default nextui-c-gEEnVX-duCSmO-textColor-default nextui-c-gEEnVX-iHugVF-size-md nextui-c-gEEnVX-krOtXB-borderWeight-normal nextui-c-gEEnVX-bAEoQm-rounded-true nextui-c-PJLV-gulvcB-isFocusVisible-false only-text-avatar"
    data-state="ready"
  >
    <span
      class="nextui-avatar-bg"
    />
    <span
      class="nextui-avatar-text"
    >
      text
    </span>
  </span>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": Array [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Avatar should work correctly with different sizes 1`] = `"<div><span class=\\"nextui-c-gEEnVX nextui-c-PJLV nextui-c-gEEnVX-qXFXw-color-default nextui-c-gEEnVX-duCSmO-textColor-default nextui-c-gEEnVX-jROyhw-size-xs nextui-c-gEEnVX-krOtXB-borderWeight-normal nextui-c-gEEnVX-bAEoQm-rounded-true nextui-c-PJLV-gulvcB-isFocusVisible-false only-text-avatar\\" data-state=\\"ready\\"><span class=\\"nextui-avatar-bg\\"></span></span><span class=\\"nextui-c-gEEnVX nextui-c-PJLV nextui-c-gEEnVX-qXFXw-color-default nextui-c-gEEnVX-duCSmO-textColor-default nextui-c-gEEnVX-bHlgBq-size-sm nextui-c-gEEnVX-krOtXB-borderWeight-normal nextui-c-gEEnVX-bAEoQm-rounded-true nextui-c-PJLV-gulvcB-isFocusVisible-false only-text-avatar\\" data-state=\\"ready\\"><span class=\\"nextui-avatar-bg\\"></span></span><span class=\\"nextui-c-gEEnVX nextui-c-PJLV nextui-c-gEEnVX-qXFXw-color-default nextui-c-gEEnVX-duCSmO-textColor-default nextui-c-gEEnVX-iHugVF-size-md nextui-c-gEEnVX-krOtXB-borderWeight-normal nextui-c-gEEnVX-bAEoQm-rounded-true nextui-c-PJLV-gulvcB-isFocusVisible-false only-text-avatar\\" data-state=\\"ready\\"><span class=\\"nextui-avatar-bg\\"></span></span><span class=\\"nextui-c-gEEnVX nextui-c-PJLV nextui-c-gEEnVX-qXFXw-color-default nextui-c-gEEnVX-duCSmO-textColor-default nextui-c-gEEnVX-lmZTtQ-size-lg nextui-c-gEEnVX-krOtXB-borderWeight-normal nextui-c-gEEnVX-bAEoQm-rounded-true nextui-c-PJLV-gulvcB-isFocusVisible-false only-text-avatar\\" data-state=\\"ready\\"><span class=\\"nextui-avatar-bg\\"></span></span><span class=\\"nextui-c-gEEnVX nextui-c-PJLV nextui-c-gEEnVX-qXFXw-color-default nextui-c-gEEnVX-duCSmO-textColor-default nextui-c-gEEnVX-ehDSPw-size-xl nextui-c-gEEnVX-krOtXB-borderWeight-normal nextui-c-gEEnVX-bAEoQm-rounded-true nextui-c-PJLV-gulvcB-isFocusVisible-false only-text-avatar\\" data-state=\\"ready\\"><span class=\\"nextui-avatar-bg\\"></span></span></div>"`;

exports[`Avatar should work with a gradient color 1`] = `"<span class=\\"nextui-c-gEEnVX nextui-c-PJLV nextui-c-gEEnVX-lcNuyd-color-gradient nextui-c-gEEnVX-duCSmO-textColor-default nextui-c-gEEnVX-iHugVF-size-md nextui-c-gEEnVX-krOtXB-borderWeight-normal nextui-c-gEEnVX-bAEoQm-rounded-true nextui-c-PJLV-gulvcB-isFocusVisible-false only-text-avatar\\" data-state=\\"ready\\"><span class=\\"nextui-avatar-bg\\"></span></span>"`;

exports[`Avatar should work with different colors 1`] = `"<div><span class=\\"nextui-c-gEEnVX nextui-c-PJLV nextui-c-gEEnVX-eeabkr-color-primary nextui-c-gEEnVX-duCSmO-textColor-default nextui-c-gEEnVX-iHugVF-size-md nextui-c-gEEnVX-krOtXB-borderWeight-normal nextui-c-gEEnVX-bAEoQm-rounded-true nextui-c-PJLV-gulvcB-isFocusVisible-false only-text-avatar\\" data-state=\\"ready\\"><span class=\\"nextui-avatar-bg\\"></span></span><span class=\\"nextui-c-gEEnVX nextui-c-PJLV nextui-c-gEEnVX-jzyGMN-color-secondary nextui-c-gEEnVX-duCSmO-textColor-default nextui-c-gEEnVX-iHugVF-size-md nextui-c-gEEnVX-krOtXB-borderWeight-normal nextui-c-gEEnVX-bAEoQm-rounded-true nextui-c-PJLV-gulvcB-isFocusVisible-false only-text-avatar\\" data-state=\\"ready\\"><span class=\\"nextui-avatar-bg\\"></span></span><span class=\\"nextui-c-gEEnVX nextui-c-PJLV nextui-c-gEEnVX-fJMMFA-color-success nextui-c-gEEnVX-duCSmO-textColor-default nextui-c-gEEnVX-iHugVF-size-md nextui-c-gEEnVX-krOtXB-borderWeight-normal nextui-c-gEEnVX-bAEoQm-rounded-true nextui-c-PJLV-gulvcB-isFocusVisible-false only-text-avatar\\" data-state=\\"ready\\"><span class=\\"nextui-avatar-bg\\"></span></span><span class=\\"nextui-c-gEEnVX nextui-c-PJLV nextui-c-gEEnVX-eCIbjl-color-warning nextui-c-gEEnVX-duCSmO-textColor-default nextui-c-gEEnVX-iHugVF-size-md nextui-c-gEEnVX-krOtXB-borderWeight-normal nextui-c-gEEnVX-bAEoQm-rounded-true nextui-c-PJLV-gulvcB-isFocusVisible-false only-text-avatar\\" data-state=\\"ready\\"><span class=\\"nextui-avatar-bg\\"></span></span><span class=\\"nextui-c-gEEnVX nextui-c-PJLV nextui-c-gEEnVX-gYTGZH-color-error nextui-c-gEEnVX-duCSmO-textColor-default nextui-c-gEEnVX-iHugVF-size-md nextui-c-gEEnVX-krOtXB-borderWeight-normal nextui-c-gEEnVX-bAEoQm-rounded-true nextui-c-PJLV-gulvcB-isFocusVisible-false only-text-avatar\\" data-state=\\"ready\\"><span class=\\"nextui-avatar-bg\\"></span></span></div>"`;

exports[`Avatar should work with pointer 1`] = `"<span class=\\"nextui-c-gEEnVX nextui-c-PJLV nextui-c-gEEnVX-qXFXw-color-default nextui-c-gEEnVX-duCSmO-textColor-default nextui-c-gEEnVX-iHugVF-size-md nextui-c-gEEnVX-krOtXB-borderWeight-normal nextui-c-gEEnVX-hpLzEY-pointer-true nextui-c-gEEnVX-bAEoQm-rounded-true nextui-c-PJLV-gulvcB-isFocusVisible-false only-text-avatar\\" data-state=\\"ready\\"><span class=\\"nextui-avatar-bg\\"></span></span>"`;

exports[`Avatar should work with zoomed 1`] = `"<span class=\\"nextui-c-gEEnVX nextui-c-PJLV nextui-c-gEEnVX-qXFXw-color-default nextui-c-gEEnVX-duCSmO-textColor-default nextui-c-gEEnVX-iHugVF-size-md nextui-c-gEEnVX-krOtXB-borderWeight-normal nextui-c-gEEnVX-bAEoQm-rounded-true nextui-c-gEEnVX-OUZAS-zoomed-true nextui-c-PJLV-gulvcB-isFocusVisible-false only-text-avatar\\" data-state=\\"ready\\"><span class=\\"nextui-avatar-bg\\"></span></span>"`;
